From 9bc747e8cd048df30987de05b67b289d40427c03 Mon Sep 17 00:00:00 2001
From: root <root@pop-os.localdomain>
Date: Sat, 1 Oct 2022 19:18:00 -0700
Subject: [PATCH] patch mxfdec to support larger kvls (seems necessary for 4k
 uncompressed j2k streams)

---
 libavformat/mxfdec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/mxfdec.c b/libavformat/mxfdec.c
index 182b122650..a5c55410ff 100644
--- a/libavformat/mxfdec.c
+++ b/libavformat/mxfdec.c
@@ -57,7 +57,7 @@
 #include "internal.h"
 #include "mxf.h"
 
-#define MXF_MAX_CHUNK_SIZE (32 << 20)
+#define MXF_MAX_CHUNK_SIZE (32 << 22)
 
 typedef enum {
     Header,
-- 
2.34.1

